### PR TITLE
#141404643 Convert from grunt to gulp

### PR DESCRIPTION
**Pull request by thebere**
##### What does this PR do?
Convert grunt to gulp task runner

##### Description of Task to be completed?
1. Livereload of development files (.js, .css, .scss, .html)
2. Linting using Eslint
3. Server start and server reload
4. Unit testing for backend
5. Testing for for frontend

##### How should this be manually tested?
**Run 'gulp' on the terminal:**
- starts sever
- performs browser reload on file changes
- performs server reload on file changes
**Run 'gulp lint' on the terminal**
- tests syntax adherence to airbnb, and specifed eslint rules
**Run 'gulp test' on the terminal**
- performs front and backend tests
**Run 'gulp test-front' on terminal**
- performs frontend test
**Run 'gulp test-front' on terminal**
- performs backend test

##### Any background context you want to provide?
N/A

##### What are the relevant pivotal tracker stories?
Convert from Grunt to Gulp task runner (#141404643)

##### Screenshots (if appropriate)
N/A

##### Questions:
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,46 +13,44 @@
...
+    "underscore": "~1.5.2",
+    "view-helpers": "~0.1.3"
   },
   "devDependencies": {
-    "supertest": "~0.8.0",
+    "browser-sync": "^2.18.8",
+    "eslint": "^3.17.1",
+    "eslint-config-airbnb-base": "^11.1.1",
+    "eslint-plugin-import": "^2.2.0",
+    "grunt": "^1.0.1",
```
Comment (sdowoade) [inline package.json]: Do we still need grunt?

**Review by sdowoade — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -13,46 +13,43 @@
     "npm": "1.2.x"
   },
   "scripts": {
-    "start": "node node_modules/grunt-cli/bin/grunt",
-    "test": "node node_modules/grunt-cli/bin/grunt test",
-    "postinstall": "node node_modules/grunt-cli/bin/grunt install"
+    "start": "gulp",
+    "test": "",
```
Comment (sdowoade) [inline package.json]: Looks like we are going to need a task for running tests.

**Review by sdowoade — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (thebere) [reply package.json]: Okay. Working on that

File: .env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+PORT= 3000
```
Comment (sdowoade) [inline .env]: This shouldn't be in here.

File: test/backend/backendSpec.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const jasmine = require('jasmine');
```
Comment (sdowoade) [inline test/backend/backendSpec.js:1]: I don't think this is required. It's not being used.

File: package.json [hunk 5]
```diff
@@ -13,48 +13,54 @@
...
+    "coveralls": "^2.12.0",
+    "dotenv": "^4.0.0",
+    "eslint": "^3.17.1",
+    "eslint-config-airbnb-base": "^11.1.1",
+    "eslint-plugin-import": "^2.2.0",
+    "gulp": "^3.9.1",
+    "gulp-eslint": "^3.0.1",
+    "gulp-jasmine": "^2.4.2",
+    "gulp-karma": "0.0.5",
+    "gulp-mocha": "^4.1.0",
```
Comment (sdowoade) [inline package.json]: Are we using mocha or jasmine?

File: .travis.yml [hunk 6]
```diff
@@ -1,8 +1,24 @@
 language: node_js
 
 node_js:
-  - node
-  - 6
+  - "6"
```
Comment (sdowoade) [inline .travis.yml:4]: Don't understand this change. Can you clarify?

**Review by sdowoade — CHANGES_REQUESTED** [review]

File: coverage/Chrome 57.0.2987 (Mac OS X 10.12.3)/lcov-report/base.css [hunk 7]
```diff
@@ -0,0 +1,213 @@
+body, html {
```
Comment (sdowoade) [inline coverage/Chrome 57.0.2987 (Mac OS X 10.12.3)/lcov-report/base.css]: The coverage folder is auto-generated after tests are run. This shouldn't be included in the repo. Please add `coverage` to the `.gitignore` file. 

You can revert the commit that included this.

**Review by sdowoade — CHANGES_REQUESTED** [review]

**Review by sdowoade — APPROVED** [review]

**Pull request merged**